transport/tchannel/handler.go

```diff
@@ -145,7 +145,9 @@ func (h handler) handle(ctx context.Context, call inboundCall) {
 			responseWriter.AddHeader(ErrorMessageHeaderKey, status.Message())
 		}
 	}
-	if err := responseWriter.Close(); err != nil {
+	// try to close response writer, however, if the caller has timed out, there is no
+	// way to write anything back (including system error).
+	if err := responseWriter.Close(); err != nil && !isTChannelTimeoutError(err) {
```

Comment:
Are we sure that the right error to be checking on is from closing the `responseWriter`? 
Do we see an error from [here](https://sourcegraph.com/github.com/yarpc/yarpc-go@5ea3960f745fb51cc6d3a99f3f7059d31d0adbb8/-/blob/api/transport/handler_invoker.go#L76) in your logs? Or is it a different log message?

Can you provide an example log output (with PII excluded of course, or pinging me internally)?

I ask, because if it's the handler returning the context deadline exceeded error (that I linked above), then we should be checking the error before the `responsewriter.Close()` call.

Reply:
I'm not seeing the error you showed. I think the reason is we wrap timeout error with layer information (in Glue):
controller.submission.submit_background_check: code:deadline-exceeded message:timeout
So err != context.DeadlineExceeded

Just pinged you the log in uChat